contracts/HashRegistrarSimplified.sol

```diff
@@ -238,7 +242,38 @@ contract Registrar {
         }
         return len;
     }
+    
+    /** 
+     * @dev Creates a scheduled release for each name
+     * 
+     * Each name will be assigned a random date in which its auction 
+     * can be started, from 0 to 6 months
+     * 
+     * @param _hash The hash to start an auction on
+     * @param _timestamp The timestamp to query about
+     */
+     
+    function isAllowed(bytes32 _hash, uint _timestamp) constant returns (bool allowed){
+        uint upperLimit;
+        upperLimit--;       // Underflows to set it to 0xFFF...
+        uint timeSpan = _timestamp - registryStarted;
+        uint limit = timeSpan < launchLength ? 
+            timeSpan * (upperLimit / launchLength)
+            : upperLimit;
+        
+        return uint(_hash) < limit;
+    }
 
+    /** 
+     * @dev Returns expected launch date for hash
+     * 
+     * Returns the minimum date for opening the auction
+     * 
+     * @param _hash The hash to start an auction on
+     */
+    function getAllowedTime(bytes32 _hash) constant returns (uint timestamp) {
+        return registryStarted + launchLength*(uint(_hash)/2**128)/2**128;
```

Comment:
How about using the left shift operators? I did a simple benchmark, the shift version costs a little bit less gas than the division-power version at 296gas vs. 335gas. Not sure if it matters for the `constant` functions.

![image](https://cloud.githubusercontent.com/assets/705367/24800981/eeb5d1ba-1bd4-11e7-9f8c-89c2b3ea15b6.png)

PS: any concern for `launchLength*(uint(_hash)/2**256)`